### ISSUE TITLE
Vault-gatekeeper-mesos will not follow 307 redirect of VAULT_ADDR

### ISSUE THREAD

**Issue by matthewmrichter** [ChannelMeter/vault-gatekeeper-mesos#2]
Vault handles high availability in active-standby mode. When a user curls a Vault node in Standby mode, it responds with a 307 redirect to the Active vault node's VAULT_ADVERTISE_ADDR: 

```
curl -i -L -H "X-Vault-Token: $VAULT_TOKEN"  -X GET $VAULT_ADDR/v1/secret/?list=true

HTTP/1.1 307 Temporary Redirect
Content-Type: text/plain; charset=utf-8
Date: Wed, 15 Jun 2016 20:36:56 GMT
Location: http://<$VAULT_ADVERTISE_ADDR>/v1/secret/?list=true
Content-Length: 0
Connection: keep-alive

HTTP/1.1 200 OK
Content-Type: application/json
Date: Wed, 15 Jun 2016 20:36:56 GMT
Content-Length: 147
Connection: keep-alive
```

I provided VGM with a VAULT_ADDR of our service discovery endpoint for Vault services. VGM doesn't seem to be following the redirects, as I am getting `{"status":"Sealed","ok":false,"error":"Error loading policy from vault: 307: communication error."}` when it attempts to communicate with the VAULT_ADDR.

I believe that a VGM should inspect a 307 response for the location and then try reaching that, in order to be compatible with a vault HA setup.

**Comment by nemothekid** [comment]
Fixed by 4cecd307c2a0cd17a58a82530d6c8306f7010abd / v0.1.3

**Comment by matthewmrichter** [comment]
Confirmed this works!
